modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/connectivity/platform/schema/ConnectivitySchema.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.connectivity.platform.schema;
+
+import org.mule.runtime.api.connection.ConnectionProvider;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Models the AML dialect used by the CFM service to represent a connection schema
+ *
+ * @since 4.4.0
+ */
+public class ConnectivitySchema {
+
+  /**
+   * Builder object for producing {@link ConnectivitySchema} instances
+   *
+   * @since 4.4.0
+   */
+  public static class Builder {
+
+    private final ConnectivitySchema product = new ConnectivitySchema();
+    private final ConnectivitySchemaNode connectionNode = new ConnectivitySchemaNode();
+
+    public Builder() {
+      product.definition.getNodeMappings().put("Connection", connectionNode);
+      uses("connectivity", "anypoint://semantics/connectivity")
+          .usesExternal("core", "anypoint://semantics/core")
+          .usesExternal("apiContract", "anypoint://semantics/api-contract");
+    }
+
+    /**
+     * Sets the schema's GAV
+     *
+     * @param groupId    a valid Maven {@code groupId}
+     * @param artifactId a valid Maven {@code artifactId}
+     * @param version    a valid Maven {@code version}
+     * @return {@code this} builder
+     */
+    public Builder setGav(String groupId, String artifactId, String version) {
```

Comment:
I would name the builder methods without the `set`.

setGav -> gav
addParameter -> withParameter